### ISSUE TITLE
Some files missing during agent installation

### ISSUE THREAD

**Issue by fhielpos** [wazuh/wazuh-bosh#14]
Hello team,

We have seen that some files and directories where missing while initializing a Wazuh Agent deployed with bosh:

Directories list:
```
/var/vcap/data/wazuh-agent/etc/
/var/vcap/packages/wazuh-agent/etc/
/var/vcap/packages/wazuh-agent/bin
```

File missing:
```
/var/vcap/packages/wazuh-agent/bin/agent-auth
```

We would need to do some further testing for this deployment to make sure that is working as intended.

Regards,
Franco Hielpos

**Comment by teddytpc1** [comment]
I have been making some changes and testing. The branch name is `14-missing-agent-files`.
I had to modify the pre-start script to add the directories creation and modify the logic of the Agent registration process.
Also had to modify the registration.erb to modify the logic of the section that checks if the Wazuh Agent is connected.
Here is some evidence:
- Agent registration log:
```
2021-09-15 19:59:35 : INFO : Registration success
2021-09-15 19:59:43 : INFO : Agent connected to the Manager.
2021-09-15 19:59:52 : INFO : Agent successfully installed, registered, and connected to the manager.
```

- List of available agents on the Manager:
```
/var/ossec/bin/agent_control -lc

Wazuh agent_control. List of available agents:
   ID: 000, Name: 88ecf36b-99f9-4356-84dc-d6db81be65f9 (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: bosh-b0a68e50-bfd7-4220-a3c1-fe127caeb1bc, IP: any, Active
```

Agent client.keys:
```
cat /var/ossec/etc/client.keys 
001 bosh-b0a68e50-bfd7-4220-a3c1-fe127caeb1bc any c687e69859bafd652437a5b392b38c0167160b3ca335c1b2302e71b082d667d7
```

Agent ossec-control status:
```
wazuh-clusterd is running...
wazuh-modulesd is running...
ossec-monitord is running...
ossec-logcollector is running...
ossec-remoted is running...
ossec-syscheckd is running...
ossec-analysisd is running...
ossec-maild not running...
ossec-execd is running...
wazuh-db is running...
ossec-authd is running...
ossec-agentlessd not running...
ossec-integratord not running...
ossec-dbd not running...
ossec-csyslogd not running...
wazuh-apid is running...
```



Agent ossec.log:
`2021/09/15 19:59:46 ossec-agentd: INFO: (4102): Connected to the server (172.31.32.4:1514/tcp).
`
Manager ossec.log:
```
2021/09/15 19:59:36 ossec-authd: INFO: New connection from 172.31.32.5
2021/09/15 19:59:36 ossec-authd: INFO: Received request for a new agent (bosh-b0a68e50-bfd7-4220-a3c1-fe127caeb1bc) from: 172.31.32.5
2021/09/15 19:59:36 ossec-authd: INFO: Agent key generated for 'bosh-b0a68e50-bfd7-4220-a3c1-fe127caeb1bc' (requested by any)
2021/09/15 19:59:37 ossec-remoted: INFO: (1409): Authentication file changed. Updating.
2021/09/15 19:59:37 ossec-remoted: INFO: (1410): Reading authentication keys file.
```

Bosh VMs status (nginx deployment contains the agent):
```
Deployment 'nginx'

Instance                                    Process State  AZ  IPs          VM CID               VM Type  Active  Stemcell  
nginx/4ff9f067-efc7-4b19-a714-1881888d8dac  running        z1  172.31.32.5  i-0eafe40fbb6bfafd7  default  true    bosh-aws-xen-hvm-ubuntu-bionic-go_agent/1.13  

1 vms

Deployment 'wazuh-manager'

Instance                                            Process State  AZ  IPs          VM CID               VM Type  Active  Stemcell  
wazuh-manager/997db35c-5fd0-4fd1-bab6-a315458e5e86  running        z1  172.31.32.4  i-012b02122efc669fb  default  true    bosh-aws-xen-hvm-ubuntu-bionic-go_agent/1.13  

1 vms
```

**Comment by teddytpc1** [comment]
Tomorrow I will check there is anything else to test and will generate the PR.

**Comment by teddytpc1** [comment]
I have tested the Wazuh Cluster installation and got the same results as in the single Manager installation. 
Will create the PR with this changes.

**Comment by teddytpc1** [comment]
Found some things to improve in the Agent deployment. I have made some additional changes and ran the tests. Everything works fine.
- Increased the sleep time in the post-start script. It validates if the Agent connected to the Manager by searching a string in the ossec.log.
- Reverted some changes I made yesterday because I had overwritten some parameter references.
- Corrected a format issue in the README.
